handler/population-types.go

```diff
@@ -35,27 +43,68 @@ func (h *PopulationTypes) Get(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// Get is the handler for GET /area-types
+/*
+   GET population-type/<id>/area-types
+   Gets area types from cantabular.
+   If not authenticated, then only return
+   area-types if datasets using it published.
+   Whether published determined by making
+   unauthenticated request to dataset api.
+*/
 func (h *PopulationTypes) GetAreaTypes(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 
+	logData := log.Data{}
+	authenticated := h.authenticate(r, logData)
+
+	isBasedOn := chi.URLParam(r, "population-type")
+
 	req := cantabular.GetGeographyDimensionsRequest{
-		Dataset: chi.URLParam(r, "population-type"),
+		Dataset: isBasedOn,
 	}
 
 	res, err := h.cantabularClient.GetGeographyDimensions(ctx, req)
 	if err != nil {
 		h.responder.Error(
 			ctx,
 			w,
-			h.cantabularClient.StatusCode(err), // Can be changed to ctblr.StatusCode(err) once added to Client
+			h.cantabularClient.StatusCode(err),
 			errors.Wrap(err, "failed to get area-types"),
 		)
 		return
 	}
 
 	var resp contract.GetAreaTypesResponse
 
+	if !authenticated {
+		datasets, err := h.datasetClient.GetDatasets(ctx,
+			"",
+			"",
+			// What do I put here for collection ID ?!
```

Comment:
Here, I was not sure where I get the collection ID from, could someone please provide advice?

Reply:
I don't think you need any `collectionID` here, as it is going to be added to the Header.
We may need the auth and service token though

Reply:
no need for auth and service, as it will be an unauthenticated request

Reply:
Will then leave this to be empty string and resolve, cheers!